### PR TITLE
UPT: Use strict compare & remove an else statment

### PR DESCRIPTION
**Pull request by AlphaWong**
Dear community,
 
Plz review my code

Best,
Alpha

### Reference Issue
https://github.com/Polymer/polymer/issues/4792

### CODE REVIEW

File: lib/elements/dom-if.html [hunk 1]
```diff
@@ -175,12 +175,13 @@
           if (!template) {
             // Wait until childList changes and template should be there by then
             let observer = new MutationObserver(() => {
-              if (this.querySelector('template')) {
-                observer.disconnect();
-                this.__render();
-              } else {
+              // If template tag not found, it will throw
```
Comment (azakus) [inline lib/elements/dom-if.html]: Why change this `if` statement?

Comment (AlphaWong) [inline lib/elements/dom-if.html]: As an early throw will be done if the first condition has been hit.
I do not think it is necessary to create an else block.

Thanks for your review.

Comment (TimvdLippe) [inline lib/elements/dom-if.html]: I would say to revert this change. It makes the code harder to read imo.

**Comment by AlphaWong** [comment]
okay, I will revert that change.

Thanks

**Comment by AlphaWong** [comment]
I just rebase to latest remote polymer and revert the change of the error throw.
Please review my code.

Thanks
Alpha

**Comment by TimvdLippe** [comment]
I think your rebase went wrong as there a couple of extra commits in this PR. Could you remove those commits and force push?

**Comment by AlphaWong** [comment]
I am not sure as I rebase to polymer/polymer rather than my personal branch.
That my commits should behind your commit (change semi quotes error level ).

I will change it tmr. 
Thanks a lot !!

**Comment by AlphaWong** [comment]
PR continue:
https://github.com/Polymer/polymer/pull/4802

Sorry for my lack experience action.
